src/OpenTelemetry.Instrumentation.Http/HttpClientInstrumentationOptions.cs

```diff
@@ -28,12 +29,48 @@ namespace OpenTelemetry.Instrumentation.Http
     public class HttpClientInstrumentationOptions
     {
         /// <summary>
-        /// Gets or sets a Filter function that determines whether or not to collect telemetry about requests on a per request basis.
-        /// The Filter gets the HttpRequestMessage, and should return a boolean.
-        /// If Filter returns true, the request is collected.
-        /// If Filter returns false or throw exception, the request is filtered out.
+        /// Gets or sets a filter function that determines whether or not to
+        /// collect telemetry about requests on a per request basis.
         /// </summary>
-        public Func<HttpRequestMessage, bool> Filter { get; set; }
+        /// <remarks>
+        /// Notes:
+        /// <list type="bullet">
+        /// <item><b>FilterHttpRequestMessage is only executed on .NET and .NET
+        /// Core runtimes. <see cref="HttpClient"/> and <see
+        /// cref="HttpWebRequest"/> on .NET and .NET Core are both implemented
+        /// using <see cref="HttpRequestMessage"/>.</b></item>
+        /// <item>The return value for the filter function is interpreted as:
+        /// <list type="bullet">
+        /// <item>If filter returns <see langword="true" />, the request is
+        /// collected.</item>
+        /// <item>If filter returns <see langword="false" /> or throws an
+        /// exception the request is NOT collected.</item>
+        /// </list></item>
+        /// </list>
+        /// </remarks>
+        public Func<HttpRequestMessage, bool> FilterHttpRequestMessage { get; set; }
+
+        /// <summary>
+        /// Gets or sets a filter function that determines whether or not to
+        /// collect telemetry about requests on a per request basis.
+        /// </summary>
+        /// <remarks>
+        /// Notes:
+        /// <list type="bullet">
+        /// <item><b>FilterHttpWebRequest is only executed on .NET Framework
+        /// runtimes. <see cref="HttpClient"/> and <see cref="HttpWebRequest"/>
+        /// on .NET Framework are both implemented using <see
+        /// cref="HttpWebRequest"/>.</b></item>
+        /// <item>The return value for the filter function is interpreted as:
+        /// <list type="bullet">
+        /// <item>If filter returns <see langword="true" />, the request is
+        /// collected.</item>
+        /// <item>If filter returns <see langword="false" /> or throws an
+        /// exception the request is NOT collected.</item>
+        /// </list></item>
+        /// </list>
+        /// </remarks>
+        public Func<HttpWebRequest, bool> FilterHttpWebRequest { get; set; }
```

Comment:
Would this be useful to not pollute the API for net6.0 folk?

```suggestion
#if NETFRAMEWORK || NETSTANDARD2_0
        public Func<HttpWebRequest, bool> FilterHttpWebRequest { get; set; }
#endif
```

Reply:
Ohh... I just realized the `net6.0` target was removed. Is it possible to keep it?